Tasks/XamariniOS/Tests/L0.ts

```diff
@@ -14,63 +14,63 @@ describe('XamariniOS L0 Suite', function () {
 
     });
 
-    it('XamariniOS signing with identifiers', (done:MochaDone) => {
+    it('XamariniOS signing with identifiers', function (done: MochaDone) {
         this.timeout(1000);
 
-        let tp = path.join(__dirname, 'L0SignWithIds.js');
-        let tr : ttm.MockTestRunner = new ttm.MockTestRunner(tp);
+        const tp = path.join(__dirname, 'L0SignWithIds.js');
+        const tr = new ttm.MockTestRunner(tp);
 
         tr.run();
 
         assert(tr.ran('/home/bin/xbuild src/project.sln /p:Configuration=Release /p:Platform=iPhone ' +
             '/p:Codesignkey=testSignIdentity /p:CodesignProvision=testUUID'),
                 'xbuild should have run with codesign for IDs');
-        assert(tr.stderr.length == 0, 'should not have written to stderr');
+        assert(tr.stderr.length === 0, 'should not have written to stderr');
         assert(tr.succeeded, 'task should have succeeded');
 
         done();
     });
 
-    it('XamariniOS signing with files', (done:MochaDone) => {
-        this.timeout(1000);
+    it('XamariniOS signing with files', function (done: MochaDone) {
+        this.timeout(2500);
```

Comment:
why increase in timeout?

Reply:
This test was timing out when I added the wildcard expansion (1700 - 1900 ms running time).  I'm not sure what caused the increase in running time, since the code that actually does wildcard expansion from vsts-task-lib should be mocked out.  Maybe whatever mechanism is intercepting the function call is just expensive.

Reply:
We should ensure the mocking is doing what we think.

Reply:
I checked this in master and it turns out the test is taking the same amount of time as before.  But the timeout wasn't actually being set before with the arrow function and was left as the default 2000 ms.  When I changed it to a classic function expression the 1000 ms timeout kicked in.

Reply:
hmmm - how does what ts transpiler created compare to the classic function expression?  Any difference if using node 8 LTS.  Just curious what the diff is.  If there's a solid reason then I'm not opposed to the classic syntax ...  just trying to understand. :)

Reply:
It looks like TS keeps whatever is in the source: `function` as `function` and `=>` as `=>`.  This makes sense because the two treat `this` differently: `this` in a `function` refers to the function object while `this` in an arrow function is bound from the enclosing scope like any other variable.  Mocha relies on the way `function` treats `this`: https://mochajs.org/#arrow-functions

Reply:
The target in the tsconfig controls the transpilation of arrow functions.  Since this repo targets ES6, arrow functions are transpiled to arrow functions because they exist in ES6 Javascript.  If it targeted ES5, they'd be transpiled into `function` declarations.

Reply:
Node 8 LTS should be fully ES6 compliant.  Do the mocha tests work if node 8 is used?

Reply:
I ran the tests on Node 8 and still don't see the timeouts taking effect.  But the issue isn't ES6 compliance; it's the different semantics of arrow functions.  I suspect Mocha is using `call` or `apply` to set the `this` context on the callback, which does not work (per the ES6 spec) with arrow functions.  Mocha was first released in 2011, so this design decision must have been made before arrow functions.

Reply:
OK - makes sense.  Thanks for running it down